packages/cardpay-sdk/sdk/constants.ts

```diff
@@ -244,16 +235,17 @@ export const networkIds: Record<string, number> = Object.freeze(
 // invert the networkIds object, so { mainnet: 1, ... } becomes { '1': 'mainnet', ... }
 export const networks: Record<string, string> = invert(networkIds);
 
-export function getConstantByNetwork<K extends ConstantKeys>(name: K, network: Network | string | number) {
-  //Convert chain id to network name
-  if (typeof network === 'number') {
-    network = networks[network];
-  }
+export function getConstantByNetwork<K extends NetworkConstantsKeys<N>, N extends Network | string>(
+  name: K,
+  network: N
+): NetworkConstants<N>[K] {
+  // @ts-expect-error ts doesn't know how to handle indexed type, but function params and return already does that;
+  const value = constants[network as Network][name];
 
-  let value = constants[network as Network][name];
-  if (value == null) {
-    throw new Error(`Don't know about the constant '${name}' for network ${network}`);
+  if (!value && value !== 0) {
```

Comment:
I've replaced this condition to explicitly show that we don't want to error on a 'zero' case